### DISCUSSION TITLE
An issue of physics plugin

### DISCUSSION THREAD

**Discussion by izzm123** [cyberbotics/webots#4820] [Q&A]
Hi, I have another problem about the physics plugin. 

Two "Solid" nodes were added to the "Robot" node, and a hinge joint was attached between this two solids by the physics plugin. As a result, this two solids penetrated the floor under the gravity mistakenly after the simulation starts.

But, while this two "Solid" nodes were added to the "Group" node, this two solids will been caught by the floor correctly. 

This two solids need to be added to the "Robot" node for they need controller.

So, could you give me some suggestions? 
Thanks!

Here is the source file with the issue, please.
[demo1.zip](https://github.com/cyberbotics/webots/files/9034130/demo1.zip)

**Comment by omichel** [comment]
Maybe a simpler approach would be to add the Solid nodes to the Robot using a supervisor process rather than a physics plugin? The supervisor API allows you to add/remove node at every level of the scene tree.

**Reply by izzm123** [reply]
Thank you!

But, some physical behavior need to be added. For example, some joints are expected to be enabled or disabled under certain conditions. This could have been realized by collision. But, the console show that the model is complex. So ,I want to realize it by some easier way in physics plugin.

I want to know whether the supervisor API is runing in realtime, or just as configuration. 

Thanks!

**Reply by omichel** [reply]
All this can be achieved by a supervisor running in real-time.

**Reply by izzm123** [reply]
OK，thank you! I will try it.
